### PR TITLE
Adding docker-compose.prod.yaml to ease deployment

### PR DESCRIPTION
**Pull request by moufmouf**
This Pr is built on top of #577 .

It:

- reverts the way the front container is built (the images at hub.docker.com must be configurable and therefore, we need to do the build on startup, even if it takes time and disk space)
- adds Let's encrypt support

### CODE REVIEW

**Comment by loelkes** [comment]
Would be open to place the traefik service in a separate compose file? I might not be the only one already running traefik or another reverse proxy.

People who want the traefik configuration can still add `-f docker-compose.traefik.yml` to their compose command. I could prepare a PR if you agree.

File: docker-compose.prod.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    image: traefik:v2.3
+    command:
+      - --log.level=WARN
+      #- --api.insecure=true
+      - --providers.docker
+      - --entryPoints.web.address=:80
+      - --entrypoints.web.http.redirections.entryPoint.to=websecure
+      - --entrypoints.web.http.redirections.entryPoint.scheme=https
+      - --entryPoints.websecure.address=:443
+      - --certificatesresolvers.myresolver.acme.email=d.negrier@thecodingmachine.com
```
Comment (djmaze) [inline docker-compose.prod.yaml:13]: The email could also be moved to the env file?

File: front/nginx-vhost.conf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+server {
```
Comment (djmaze) [inline front/nginx-vhost.conf:1]: This file seems unused?

**Comment by djmaze** [comment]
In my experience docker-based production deployments tend to be very opinionated and specific to one's needs. Maybe move this configuration into an *examples* folder or *contrib/docker*? Then we could have different example configurations, e.g. for different reverse proxies.

Just my 2 cents.

File: docker-compose.prod.yaml (hunk 1 above)
Comment (moufmouf) [reply docker-compose.prod.yaml:13]: Woopsie :) Thanks!

File: front/nginx-vhost.conf (hunk 2 above)
Comment (moufmouf) [reply front/nginx-vhost.conf:1]: Indeed! Thanks!

**Comment by moufmouf** [comment]
@djmaze Thanks for your review. I like a lot the idea of putting this in a `contrib/docker` folder and being explicit about this file being quite opiniated and only a starting point. :+1:

**Pull request merged**